### PR TITLE
Make sure savedInstanceState has accurate isSearchRequested value

### PR DESCRIPTION
**Pull request by clauz9**
### First time contributor checklist

- [x] I have read [how to contribute](https://github.com/signalapp/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/signalapp/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Nokia 3.1, Android 10
 * Samsung A10, Android 11
 * Huawei L29, Android 8
 * Samsung S5 mini, Android 5.1
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description

fixes signalapp#12054

[`isSearchRequested` was not being set to true when searchViewItem's actionView was expanded](https://github.com/signalapp/Signal-Android/blob/cba784b8ec14fa58a550e845d383395945bb7d00/app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationParentFragment.java#L1065-L1069), which leads to [`onSavedInstanceState` saving the wrong value for `isSearchRequested`,](https://github.com/signalapp/Signal-Android/blob/cba784b8ec14fa58a550e845d383395945bb7d00/app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationParentFragment.java#L859-L864) and then the app is not aware that the search was open before onStop() was called. 

I also used the onBackPressed() method to make sure that if search is opened when back is pressed it first closes search, and returns the `inputPanel` instead of finishing the activity.

### CODE REVIEW

**Comment by alex-signal** [comment]
Is this fixing the issue where the search UI doesn't reappear properly?

**Comment by clauz9** [comment]
> Is this fixing the issue where the search UI doesn't reappear properly?

yes. the first commit, 

I will be adding a video of the behaviors before and after,  

there's also another issue, 
when quoting a message when the search UI is open, the handleReply method is called but the search UI is not closed, I'm pushing a commit to check for `isSearchRequested` in `handleReplyMessage()`

**Comment by alex-signal** [comment]
Ok, please leave a comment when you're ready for a review.

**Comment by clauz9** [comment]
It should be ready, Alex
I've also added a video of the changed behavior. 

https://user-images.githubusercontent.com/77681918/158060795-78f43912-b4c3-4327-8ef8-8c1fbe85630e.mp4

File: app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationParentFragment.java [hunk 1]
```diff
@@ -3820,6 +3823,8 @@ public void setThreadId(long threadId) {
 
   @Override
   public void handleReplyMessage(ConversationMessage conversationMessage) {
+    if (isSearchRequested) searchViewItem.collapseActionView();
```
Comment (alex-signal) [inline app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationParentFragment.java:3826]: Please use brackets and a new line

**Review by alex-signal — APPROVED** [review]

**Comment by clauz9** [comment]
I just got home, is this ok ?

**Comment by cody-signal** [comment]
Merged into next beta release

**Pull request closed without merging**